module/VuFindTheme/src/VuFindTheme/ResourceContainer.php

```diff
@@ -129,12 +129,101 @@ public function addCss($css)
      */
     public function addJs($js)
     {
-        if (!is_array($js) && !is_a($js, 'Traversable')) {
-            $js = [$js];
+        if ((!is_array($js) && !is_a($js, 'Traversable')) || isset($js['file'])) {
+            $this->addJsEntry($js);
+        } elseif (isset($js[0])) {
+            foreach ($js as $current) {
+                $this->addJsEntry($current);
+            }
+        } elseif ($js == []) {
+            return;
+        } else {
+            throw new \Exception("Invalid JS entry format: " . print_r($js, true));
+        }
+    }
+
+    /**
+     * Helper function for adding a Javascript file.
+     *
+     * @param string|array $jsEntry Entry to add, either as string with path
+     * or array with additional properties.
+     *
+     * @return void
+     */
+    protected function addJsEntry($jsEntry)
+    {
+        if (!is_array($jsEntry)) {
+            $parts = $this->parseSetting($jsEntry);
+            if (count($parts) == 1) {
+                $jsEntry = ['file' => $jsEntry];
+            } else {
+                $jsEntry = [
+                    'file' => $parts[0],
+                    'attributes' => ['conditional' => trim($parts[1])],
+                ];
+            }
+        } elseif (isset($jsEntry['priority']) && isset($jsEntry['load_after'])) {
+            throw new \Exception(
+                'Using "priority" as well as "load_after" in the same entry '
+                . ' is not supported: "' . $jsEntry['file'] . '"'
+            );
+        }
+
+        if (!isset($jsEntry['position'])) {
+            $jsEntry['position'] = 'header';
+        }
+
+        foreach ($this->js as $existingEntry) {
+            if ($existingEntry['file'] == $jsEntry['file']) {
```

Comment:
Does this mean that if I inherit a theme and add the same js as the parent theme in my child theme, it will blow up? If so, that will break the current behavior for us. We list all the scripts in order for inclusion in asset pipeline.

Reply:
I think this probably will cause a problem, then. Do we need some kind of a deeper comparison function so it only fails if the filename is the same but other details are different?

Reply:
The code will now use array_diff and only throw an Exception if the new setting differs from the old one.